packages/graph/hash_graph/lib/graph/src/datastore/postgres/mod.rs

```diff
@@ -372,102 +360,184 @@ impl PostgresDatabase {
+            transaction
+                .fetch_one(
+                    sqlx::query(
+                        r#"
+                    INSERT INTO property_type_data_type_references (source_property_type_version_id, target_data_type_version_id)
+                    VALUES ($1, $2)
+                    RETURNING source_property_type_version_id;
+                    "#,
+                    )
+                        .bind(property_type.version_id())
+                        .bind(target_id),
+                )
+                .await
+                .report()
+                .change_context(InsertionError)?;
+        }
 
         Ok(())
     }
 
     async fn insert_entity_references(
-        &self,
-        entity_type: &EntityType,
+        transaction: &mut Transaction<'_, Postgres>,
+        entity_type: &Qualified<EntityType>,
     ) -> Result<(), InsertionError> {
         // TODO: Store this as mapping in `entity_type_property_types`
-        let _property_type_ids = self
-            .property_type_reference_ids(entity_type.property_type_references())
-            .await
-            .change_context(InsertionError)
-            .attach_printable("Could not find referenced property types")?;
+        let property_type_ids = Self::property_type_reference_ids(
+            transaction,
+            entity_type.inner().property_type_references(),
+        )
+        .await
+        .change_context(InsertionError)
+        .attach_printable("Could not find referenced property types")?;
+
+        for target_id in property_type_ids {
+            transaction
+                .fetch_one(
+                    sqlx::query(
+                        r#"
+                    INSERT INTO entity_type_property_types (source_entity_type_version_id, target_property_type_version_id)
+                    VALUES ($1, $2)
+                    RETURNING source_entity_type_version_id;
+                    "#,
+                    )
+                        .bind(entity_type.version_id())
+                        .bind(target_id),
+                )
+                .await
+                .report()
+                .change_context(InsertionError)?;
+        }
 
         // TODO: Store link references
 
         Ok(())
     }
 
     async fn property_type_reference_ids<'p, I>(
-        &self,
+        transaction: &mut Transaction<'_, Postgres>,
         property_type_references: I,
     ) -> Result<Vec<VersionId>, QueryError>
     where
         I: IntoIterator<Item = &'p PropertyTypeReference> + Send,
         I::IntoIter: Send,
     {
-        futures::future::try_join_all(
-            property_type_references
-                .into_iter()
-                .map(|reference| self.version_id_by_uri(reference.uri())),
-        )
-        .await
+        let property_type_references = property_type_references.into_iter();
+        let mut ids = Vec::with_capacity(property_type_references.size_hint().0);
+        for reference in property_type_references {
```

Comment:
I see that futures has been removed as a dependency as well, what is the reason for switching away from the `join_all`? 
Wouldn't this version be "slower" as we are awaiting at each element?

Reply:
It's because of the `Transaction` struct. It requires to be `&mut`, thus it can't be done in parallel.